frontend/pages/SoftwarePage/SoftwareTitleDetailsPage/ViewYamlModal/helpers.tsx

```diff
@@ -0,0 +1,181 @@
+import React, { MouseEvent } from "react";
+
+import Button from "components/buttons/Button";
+
+interface RenderYamlHelperText {
+  installScript?: string;
+  uninstallScript?: string;
+  preInstallQuery?: string;
+  postInstallScript?: string;
+  onClickPreInstallQuery?: (evt: MouseEvent) => void;
+  onClickInstallScript?: (evt: MouseEvent) => void;
+  onClickPostInstallScript?: (evt: MouseEvent) => void;
+  onClickUninstallScript?: (evt: MouseEvent) => void;
+}
+
+export const renderYamlHelperText = ({
+  preInstallQuery,
+  installScript,
+  postInstallScript,
+  uninstallScript,
+  onClickPreInstallQuery,
+  onClickInstallScript,
+  onClickPostInstallScript,
+  onClickUninstallScript,
+}: RenderYamlHelperText): JSX.Element => {
+  const items: { key: string; element: JSX.Element }[] = [];
+
+  if (preInstallQuery) {
+    items.push({
+      key: "pre-install-query",
+      element: (
+        <Button key="pre" variant="text-link" onClick={onClickPreInstallQuery}>
+          pre-install query
+        </Button>
+      ),
+    });
+  }
+  if (installScript) {
+    items.push({
+      key: "install-script",
+      element: (
+        <Button
+          key="install"
+          variant="text-link"
+          onClick={onClickInstallScript}
+        >
+          install script
+        </Button>
+      ),
+    });
+  }
+  if (uninstallScript) {
+    items.push({
+      key: "uninstall-script",
+      element: (
+        <Button
+          key="uninstall"
+          variant="text-link"
+          onClick={onClickUninstallScript}
+        >
+          uninstall script
+        </Button>
+      ),
+    });
+  }
+  if (postInstallScript) {
+    items.push({
+      key: "post-install-script",
+      element: (
+        <Button
+          key="post"
+          variant="text-link"
+          onClick={onClickPostInstallScript}
+        >
+          post-install script
+        </Button>
+      ),
+    });
+  }
+
+  if (items.length === 0) return <></>;
+
+  // Helper to join items with commas and Oxford comma before "and"
+  const joinWithCommasAnd = (
+    elements: { key: string; element: JSX.Element }[]
+  ) => {
+    return elements.map((item, idx) => {
+      if (idx === 0) return item.element;
+      if (idx === elements.length - 1) {
+        return (
+          <React.Fragment key={`and-${item.key}`}>
+            {elements.length > 2 ? "," : ""} and {item.element}
+          </React.Fragment>
+        );
+      }
+      return (
+        <React.Fragment key={`comma-${item.key}`}>
+          , {item.element}
+        </React.Fragment>
+      );
+    });
+  };
```

Comment:
Nit: this can be moved out of the renderYamlHelpText so its not recreated every time that function is called.